### DISCUSSION TITLE
Large component storage

### DISCUSSION THREAD

**Discussion by mpeterdev** [NearSocial/viewer#109] [General]
Is it likely that some developers will create components with source large enough that it would be prohibitive to store in SocialDB?

If so, could we leverage a storage service like IPFS for the code itself then keep the CID on-chain?

**Comment by mpeterdev** [comment]
@evgenykuzyakov 

I'd like to hear your thoughts on this

**Reply by evgenykuzyakov** [reply]
It's possible to do already, but unlikely to be needed. You can pass raw code to the `<Widget />` component using `code` attr instead of `src`. So you can make a widget that fetches from IPFS and then displays it
